### ISSUE TITLE
Cannot change subnet_id on netapp-cloudmanager_connector_gcp for shared VPC

### ISSUE THREAD

**Issue by bryanheo** [NetApp/terraform-provider-netapp-cloudmanager#95]
We are deploying the connector on shared VPC in GCP as shown below but subnet_id is not changed
We have to use host1 subnet but it seems there is no way to change the subnet. We also tried hard-coded value but it did not work.
Could you have a look at the issue?

It should show:
```
  subnetwork: projects/host1/regions/us-east4/subnetworks/dev3
```

but it returns:
```
  subnetwork: projects/dev1/regions/us-east4/subnetworks/projects/ap-engg-develop-host1/regions/us-east4/subnetworks/dev3
```

Code:
```
provider "google" {
  project = "dev1"
  region  = "us-east4"
}

provider "google" {
  alias   = "shared"
  project = "host1"
  region  = "us-east4"
}

data "google_compute_subnetwork" "this" {
  provider = google.shared  
  name    = "dev3"
  region  = "us-east4"
  project = "host1"  
}

resource "netapp-cloudmanager_connector_gcp" "this" {
  depends_on            = [google_compute_firewall.this]
  name                  = "netapp-dev-connector"
  zone                  = "us-east4-c"
  company               = "xxxxx"
  service_account_email = google_service_account.deploy.email
  service_account_key   = base64decode(google_service_account_key.deploy.private_key)
  account_id            = "xxxxx"
  project_id            = "xxxxx"
  associate_public_ip   = false
  subnet_id             = data.google_compute_subnetwork.this.id
}
```

**Comment by chuyich** [comment]
google_compute_subnetwork ID is projects/{{project}}/regions/{{region}}/subnetworks/{{name}} 
cloudmanager compose the subnetwork for the users: projects/<project_id>/regions/<zone>/subnetworks/<subnet_id>
Please pass the name of the subnet.

**Comment by bryanheo** [comment]
Thank you for your update.
We have tried the following but we need to deploy it into shared VPC (projects/host1/regions/us-east4/subnetworks/dev3) rather than local vpc (projects/dev1/regions/us-east4/subnetworks/dev3). We cannot change the project from dev1 to host 1
```
subnet_id             = data.google_compute_subnetwork.this.name
```

**Comment by bryanheo** [comment]
Project should be dev1 but subnet should be host1 because of shared VPC
We tried it manually through Cloud Manager UI and it shows correctly but it seems that there is no way to change subnet_id with different project (projects/host1/regions/us-east4/subnetworks/dev3)

**Comment by wenjun666** [comment]
We assume the project_id is the project id to use when specifying the subnetwork in GCP, but apparently that's not always the case. we will try to fix it so it can work with shared VPC.

**Comment by bryanheo** [comment]
Thank you

**Comment by bryanheo** [comment]
@wenjun666 @chuyich could you let me know ETA for the issue fixed?
We need to provision multiple NetApp CVOs for production soon.

**Comment by wenjun666** [comment]
@bryanheo I couldn't give out any promise because it's tricky to test if a connector resource is involved. I will aim for a fix release in about 2 weeks, you can check back if there isn't an update by that time.

**Comment by wenjun666** [comment]
This is fixed with v22.4.0

**Comment by bryanheo** [comment]
Thank you @wenjun666 
Has subnet_id issue been fixed on the both netapp-cloudmanager_connector_gcp and netapp-cloudmanager_cvo_gcp?

**Comment by wenjun666** [comment]
@bryanheo It hasn't been fixed. I am going to reopen the issue.

**Comment by bryanheo** [comment]
Could you let me know the update? Has the issue been fixed on 22.8.0?

**Comment by wenjun666** [comment]
Both gcp_connector and gcp_cvo are fixed.

in connector:
use network_project_id as the project_id that the vpc located. and subnet_id as the name of the vpc.

in cvo:
subnet_id = "projects/<project-id>/regions/<region>/subnetworks/<subnet>"
vpc_id is the name of the vpc.
You can see it from the API request on the UI.
